### PR TITLE
Move to the spec the cutscenes of Bg_Treemough and En_Zl1

### PR DESCRIPTION
**Pull request by AngheloAlf**

### CODE REVIEW

**Review by fig02 — APPROVED** [review]

**Review by EllipticEllipsis — APPROVED** [review]

**Review by Dragorn421 — APPROVED** [review]

File: src/overlays/actors/ovl_Bg_Treemouth/z_bg_treemouth.c [hunk 1]
```diff
@@ -26,7 +26,10 @@ void func_808BC8B8(BgTreemouth* this, GlobalContext* globalCtx);
 void func_808BC9EC(BgTreemouth* this, GlobalContext* globalCtx);
 void func_808BCAF0(BgTreemouth* this, GlobalContext* globalCtx);
 
-#include "z_bg_treemouth_cutscene_data.c" EARLY
+extern CutsceneData D_808BCE20[];
+extern CutsceneData D_808BD2A0[];
+extern CutsceneData D_808BD520[];
+extern CutsceneData D_808BD790[];
```
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Treemouth/z_bg_treemouth.c:32]: I think it's out of the scope of this PR but would we want to have these symbols in a `z_bg_treemouth_cutscene_data.h` header file

Thinking ahead about extraction (?) and tooling it seems to make sense

(also github ate this comment in my approval review)

Comment (Roman971) [reply src/overlays/actors/ovl_Bg_Treemouth/z_bg_treemouth.c:32]: I agree this is a bit out of scope so we should probably discuss it in discord

**Review by Roman971 — APPROVED** [review]

**Comment by fig02** [comment]
So zl1 has a bunch of Camera float data in it that isnt really relevant to the cutscene script. Imo we should probably move this back to the overlay and have the separately linked file be just the cs script. But curious what others think

**Comment by Roman971** [comment]
I don't really have an opinion about this, so I'm fine with either. But if our plan is to have all cutscenes in their own file then I agree with moving it back to the overlay or at least another file.

**Comment by fig02** [comment]
My idea mainly is that a tool generates the cutscene script and links it as its own thing. Obviously we know this doesnt hold true in the whole codebase, cause some actors have it in the middle of their data section without a file boundary. But I think it makes sense to hold that organizational structure when possible

would be fine with a seperate file for the float data as well if people felt it was too intrusive. But for an already pretty irrelevant actor I dont really care too much either way

**Comment by Roman971** [comment]
Yeah moving it back to the overlay seems fine to me in this case

**Pull request merged**